### ISSUE TITLE
Productization / All CircleCI tests failed for 2.2.11.SP1

### ISSUE THREAD

**Issue by jacobdotcosta** [snowdrop/team#354]
All the CircleCI tests failed for `2.2.11.SP1`.

**Comment by jacobdotcosta** [comment]
At first some failed with concurrency errors others failed showing the following error, but now almost all are failing with this error.
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 53.373 s <<< FAILURE! - in dev.snowdrop.example.OpenShiftIT
[ERROR] dev.snowdrop.example.OpenShiftIT  Time elapsed: 53.371 s  <<< ERROR!
java.lang.RuntimeException: java.lang.IllegalStateException: Environment not initialized in time.
Caused by: java.lang.IllegalStateException: Environment not initialized in time.
Caused by: io.fabric8.kubernetes.clnt.v4_0.KubernetesClientTimeoutException: Timed out waiting for [480000] milliseconds for multiple resources. Resources that are not ready: [Kind:DeploymentConfig Name:health-check Namespace:fmp]
```

**Comment by jacobdotcosta** [comment]
ConfigMap example did succeed at the 2nd try - https://app.circleci.com/pipelines/github/snowdrop/configmap-example/446/workflows/baec48f4-a91b-4da3-84f0-ef25f65d1e9a/jobs/409

**Comment by jacobdotcosta** [comment]
Cache Example for Beta1, which succedded on the 3rd of December also fails with the same error without apparent changes to the code.

Job 479, for Beta 1 on the 3rd of December (SUCCESS) - https://app.circleci.com/pipelines/github/snowdrop/cache-example/485/workflows/d50b1169-821e-44fc-b860-2e3f8be2d654/jobs/479
Job 491, for Beta 1 on the 28th of December (FAILURE) https://app.circleci.com/pipelines/github/snowdrop/cache-example/485/workflows/bea99d38-ef75-4067-9583-69bda3e3bcc1/jobs/491

**Comment by jacobdotcosta** [comment]
After several tests...
```bash
$ oc -n s2i get pods
NAME           READY     STATUS    RESTARTS   AGE
crud-1-build   0/1       Error     0          26m
```

```bash
$ oc -n s2i logs crud-1-build
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  48.821 s
[INFO] Finished at: 2020-12-28T17:36:34Z
[INFO] ------------------------------------------------------------------------
WARN ARTIFACT_COPY_ARGS is deprecated.  Please use S2I_SOURCE_DEPLOYMENTS_FILTER to specify artifact types and MAVEN_S2I_ARTIFACT_DIRS to specify the build output directories to copy from.
'crud-2.2.11-4-SNAPSHOT.jar' -> '/deployments/crud-2.2.11-4-SNAPSHOT.jar'

Pushing image 172.30.1.1:5000/s2i/crud:BOOSTER_VERSION ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Registry server Address:
Registry server User Name: serviceaccount
Registry server Email: serviceaccount@example.org
Registry server Password: <<non-empty>>
error: build error: Failed to push image: After retrying 6 times, Push image still failed due to error: Get http://172.30.1.1:5000/v2/: dial tcp 172.30.1.1:5000: getsockopt: connection refused
```

**Comment by jacobdotcosta** [comment]
The rest example was showing an hibernate error on it's build...so I'm updating the example properties file with the missing hibernate dialect.

```bash
$  oc -n s2i get pods
NAME                        READY     STATUS             RESTARTS   AGE
crud-1-build                0/1       Completed          0          22m
rest-crud-example-1-xxstb   0/1       CrashLoopBackOff   8          20m
```

```bash
$ oc -n s2i logs -f rest-crud-example-1-xxstb
...
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Invocation of init method failed; nested exception is org.hibernate.service.spi.ServiceException: Unable to create requested service [org.hibernate.engine.jdbc.env.spi.JdbcEnvironment]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1794) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:594) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:516) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:324) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:322) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1109) ~[spring-context-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:869) ~[spring-context-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:551) ~[spring-context-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) ~[spring-boot-2.2.11.RELEASE.jar!/:2.2.11.RELEASE]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:747) [spring-boot-2.2.11.RELEASE.jar!/:2.2.11.RELEASE]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:405) [spring-boot-2.2.11.RELEASE.jar!/:2.2.11.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) [spring-boot-2.2.11.RELEASE.jar!/:2.2.11.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) [spring-boot-2.2.11.RELEASE.jar!/:2.2.11.RELEASE]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) [spring-boot-2.2.11.RELEASE.jar!/:2.2.11.RELEASE]
        at dev.snowdrop.example.ExampleApplication.main(ExampleApplication.java:26) [classes!/:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_272]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_272]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_272]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_272]
        at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) [crud-2.2.11-4-SNAPSHOT.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) [crud-2.2.11-4-SNAPSHOT.jar:na]
        at org.springframework.boot.loader.Launcher.launch(Launcher.java:51) [crud-2.2.11-4-SNAPSHOT.jar:na]
        at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:52) [crud-2.2.11-4-SNAPSHOT.jar:na]
Caused by: org.hibernate.service.spi.ServiceException: Unable to create requested service [org.hibernate.engine.jdbc.env.spi.JdbcEnvironment]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:275) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:237) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:214) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.id.factory.internal.DefaultIdentifierGeneratorFactory.injectServices(DefaultIdentifierGeneratorFactory.java:152) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.injectDependencies(AbstractServiceRegistryImpl.java:286) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:243) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:214) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.boot.internal.InFlightMetadataCollectorImpl.<init>(InFlightMetadataCollectorImpl.java:176) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:118) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.metadata(EntityManagerFactoryBuilderImpl.java:1224) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:1255) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.springframework.orm.jpa.vendor.SpringHibernateJpaPersistenceProvider.createContainerEntityManagerFactory(SpringHibernateJpaPersistenceProvider.java:58) ~[spring-orm-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:365) ~[spring-orm-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.buildNativeEntityManagerFactory(AbstractEntityManagerFactoryBean.java:391) ~[spring-orm-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:378) ~[spring-orm-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.afterPropertiesSet(LocalContainerEntityManagerFactoryBean.java:341) ~[spring-orm-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1853) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1790) ~[spring-beans-5.2.10.RELEASE.jar!/:5.2.10.RELEASE]
        ... 24 common frames omitted
Caused by: org.hibernate.HibernateException: Access to DialectResolutionInfo cannot be null when 'hibernate.dialect' not set
        at org.hibernate.engine.jdbc.dialect.internal.DialectFactoryImpl.determineDialect(DialectFactoryImpl.java:100) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.engine.jdbc.dialect.internal.DialectFactoryImpl.buildDialect(DialectFactoryImpl.java:54) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:137) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:35) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.initiateService(StandardServiceRegistryImpl.java:101) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:263) ~[hibernate-core-5.4.22.Final.jar!/:5.4.22.Final]
        ... 41 common frames omitted
```

**Comment by jacobdotcosta** [comment]
After this problem is solved there's an odd behavior with the database. By the time the test is executed a new database connection appears which terminates the previous one, and the new one also fails. This means no database is present for the test and it fails now with connection errors such:

* `org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to obtain JDBC Connection; nested exception is java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30001ms.`
* `Caused by: java.net.SocketTimeoutException: connect timed out`
* `Caused by: org.postgresql.util.PSQLException: The connection attempt failed.`
* `Caused by: java.sql.SQLException: HikariPool-1 - Interrupted during connection acquisition`
* `Caused by: java.lang.InterruptedException: null`

**Comment by jacobdotcosta** [comment]
The previous problem seems to be related to a second connection being created.
In the logs the following message appears:
> ```
...
Found dependency: file:/home/circleci/project/target/test-classes/database.yml
Processing template. No parameters file has been specified, processing without external parameters!
Did not find any kubernetes/openshift configuration files before starting the test execution. If you are using fabric8-maven-plugin, ensure `mvn package fabric8:resource fabric8:build` is run first to generate the resources.
2020-12-29 12:08:55.052  WARN   --- [.0.0.1:8443/...] i.f.k.c.v.d.i.WatchConnectionManager     : Exec Failure: HTTP 500, Status: 500 - invalid upgrade response: status code 200
java.net.ProtocolException: Expected HTTP 101 response but was '500 Internal Server Error'
...
```

There's the `src/test/resources/database.yml` file which contains configuration to a new database connection. I'll investigate why there's a 2nd db connection and why it is failing to start.

**Comment by jacobdotcosta** [comment]
I suspect the problem might be the database port which is defined as the same for both definitions. But why are there 2 database definitions?

**Comment by jacobdotcosta** [comment]
It's odd because this configuration has been present for a while without any problems. Maybe a new CircleCI limitation?

**Comment by jacobdotcosta** [comment]
Postponed for now...
